Makefile

```diff
@@ -16,6 +16,12 @@ uninstall:
 
 .PHONY: bench
 bench:
+	@ dune build bench/bench.exe
```

Comment:
Alright we pinpointed the problem: running `dune build ... | dune exec ...` is actually a really bad idea since both dune commands run in parallel which isn't what we want since the second one relies on the first.
A magic solution would be to put `(action (pipe-stdout (run bench...) (run conversions...))`, but `pipe-stdout` is only available since dune 2.7 or so, and we didn't want to bump your dune-project that far
A dirty solution is to run the commands sequentially, with a temp file to buffer between them...

Reply:
I have no issue bumping dune to 2.7. Fortunately dune is pretty much backward compatible to the lowest version we support so it is not that bad as other packages. The only downside is that this will also affect the minimum dune version of the main package but oh well, so be it. It if turns out to be a problem we can still go for a hackier solution later.

Reply:
I bumped to 2.7 in `dune-version` and fixed compatibility. This solves the dirtiness of the makefile by a lot.
Should I also bump the requirement in `yojson-bench.opam`?

Reply:
Yes, please do. Both opam files have to be updated to depend on dune >= 2.7.

(I will probably enable opam file autogeneration after this PR so this will be taken care of automatically in the future but let's not conflate multiple changes in one PR, this one is self-contained and good as-is)